src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -769,6 +771,252 @@ protected bool IsAlive
+            return parent;
+        }
+
+        private static Autodesk.Revit.DB.Element GetParentElementFromRailings(Autodesk.Revit.DB.Element element)
+        {
+            Autodesk.Revit.DB.Element parent;
+            var railingElement = element as Autodesk.Revit.DB.Architecture.ContinuousRail;
+            ElementId hostId = railingElement.HostRailingId;
+            if (hostId == null)
+                throw new InvalidOperationException(Properties.Resources.NoParentElement);
+            parent = Document.GetElement(hostId);
+            return parent;
+        }
+
+        private static Autodesk.Revit.DB.Element GetParentElementFromStructuralFraming(Autodesk.Revit.DB.Element element)
+        {
+            Autodesk.Revit.DB.Element parent;
+            var beam = element as Autodesk.Revit.DB.FamilyInstance;
+            if (beam == null)
+                throw new InvalidOperationException(Properties.Resources.NoParentElement);
+            parent = BeamSystem.BeamBelongsTo(beam);
+            return parent;
+        }
+
+        private static Autodesk.Revit.DB.Element GetParentComponentFromStairsLandings(Autodesk.Revit.DB.Element element)
+        {
+            Autodesk.Revit.DB.Element parent;
+            var stairElement = element as Autodesk.Revit.DB.Architecture.StairsLanding;
+            if (stairElement == null)
+                throw new InvalidOperationException(Properties.Resources.NoParentElement);
+            parent = stairElement.GetStairs();
+            return parent;
+        }
+
+        /// <summary>
+        /// Transforms this Element from source CoordinateSystem to new context CoordinateSystem.
+        /// </summary>
+        /// <param name="fromCoordinateSystem">Source CoordinatSystem</param>
+        /// <param name="contextCoordinateSystem">Context CordinateSystem</param>
+        /// <returns>Transformed Element</returns>
+        public Element Transform(CoordinateSystem fromCoordinateSystem, CoordinateSystem contextCoordinateSystem)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+            var transManager = TransactionManager.Instance.TransactionWrapper;
+            var t = transManager.StartTransaction(Document);
+            try
+            {
+                SetLocationFromCS(this, fromCoordinateSystem, contextCoordinateSystem);
+                RotateElementFromCS(this, fromCoordinateSystem, contextCoordinateSystem);
+                t.CommitTransaction();
+            }
+            catch (Exception ex)
+            {
+                t.CancelTransaction();
+                throw new Exception(ex.Message);
+            }         
+            TransactionManager.Instance.TransactionTaskDone();
+            return this;
+        }
+
+        private void RotateElementFromCS(Element element, CoordinateSystem fromCS, CoordinateSystem contextCS)
+        {
+            var familyInstance = element as FamilyInstance;
```

Comment:
It seems that the transform operation can only be performed when the element can be as a FamilyInstance.
Is this node suitable for Element? I use other Element for transform operation and only get "Element.Transform operation failed.
element" information, which does not help people use it properly

Reply:
yeah agree, this either needs better docs/error message, a more constrained type input, or both.

Reply:
I haven't been able to find a way to get the Transform of an Element, which is why I am casting it to FamilyInstance as this has a GetTransform(). Happy to hear if you guys know of a way I can get the Transform of an Element.

Maybe this should be a node under FamilyInstance instead to avoid the confusion?